src/edit.js

```diff
@@ -1,19 +1,36 @@
-import countries from '../assets/countries.json';
-import './editor.scss';
+/**
+ * WordPress dependencies
+ */
 import { edit, globe } from '@wordpress/icons';
-import { BlockControls } from '@wordpress/block-editor';
-import { ComboboxControl, Placeholder, ToolbarButton, ToolbarGroup } from '@wordpress/components';
-import { getEmojiFlag } from './utils';
+import { BlockControls, useBlockProps } from '@wordpress/block-editor';
+import {
+	ComboboxControl,
+	Placeholder,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
 import { useEffect, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import countries from '../assets/countries.json';
+import { getEmojiFlag } from './utils';
 import Preview from './preview';
+import './editor.scss';
 
 export default function Edit( { attributes, setAttributes } ) {
 	const { countryCode, relatedPosts } = attributes;
-	const options = Object.keys(countries).map(code => ({
-        value: code,
-        label:  getEmojiFlag( code ) + '  ' + countries[code] + ' — ' + code
-	}));
+
+	const blockProps = useBlockProps( {
+		countryCode,
+		relatedPosts,
+	} );
+
+	const options = Object.keys( countries ).map( ( code ) => ( {
+		value: code,
+		label: getEmojiFlag( code ) + '  ' + countries[ code ] + ' — ' + code,
+	} ) );
 
 	const [ isPreview, setPreview ] = useState();
```

Comment:
NIT: A common convention related to the `useState` hook is to add a `set` prefix to the state variable name. So if the variable name is `isPreview`, the setter should be `setIsPreview`:

```suggestion
	const [ isPreview, setIsPreview ] = useState();
```

Reply:
Later, down in the code, we use the `setPreview` setter inside a `useEffect` hook. There's a React code smell tied to that code (can't comment on the actual code fragment since it was not modified):

```js
useEffect( () => setPreview( countryCode ), [ countryCode ] );
```

The `useEffect`'s hook callback should only return a side effect cleanup function. In this case, the fat arrow function returns the setter and not a cleanup function. Since we don't need any cleanup after this effect, we should nothing from the callback, i.e.:

```js
useEffect( () => {
	setPreview( countryCode )
}, [ countryCode ] );
```